spec/features/idv/doc_auth/document_capture_spec.rb

```diff
@@ -298,9 +308,353 @@
         end
 
         context 'selfie with error is uploaded' do
+          before do
+            allow(IdentityConfig.store).to receive(:doc_auth_max_attempts).and_return(99)
+
+            allow_any_instance_of(FederatedProtocols::Oidc).
+              to receive(:biometric_comparison_required?).
+              and_return(true)
+            perform_in_browser(:mobile) do
+              visit_idp_from_oidc_sp_with_ial2(biometric_comparison_required: true)
+              sign_in_and_2fa_user(@user)
+              complete_doc_auth_steps_before_document_capture_step
+            end
+          end
+
+          it 'shows the correct error message for the given error' do
+            # when the only error is a doc auth error
+
+            perform_in_browser(:mobile) do
+              attach_images(
+                Rails.root.join(
+                  'spec', 'fixtures',
+                  'ial2_test_credential_doc_auth_fail_selfie_pass.yml'
+                ),
+              )
+              attach_selfie(
+                Rails.root.join(
+                  'spec', 'fixtures',
+                  'ial2_test_credential_doc_auth_fail_selfie_pass.yml'
+                ),
+              )
+
+              submit_images
+
+              h1_error_message = strip_tags(t('errors.doc_auth.rate_limited_heading'))
+              expect(page).to have_content(h1_error_message)
+
+              body_error_message = strip_tags(t('doc_auth.errors.dpi.top_msg'))
+              expect(page).to have_content(body_error_message)
+
+              click_try_again
+              expect(page).to have_current_path(idv_document_capture_path)
+
+              inline_error_message = strip_tags(t('doc_auth.errors.dpi.failed_short'))
+              expect(page).to have_content(inline_error_message)
+
+              expect(page).to have_current_path(idv_document_capture_url)
+
+              # when doc auth result passes but liveness fails
+
+              attach_images(
+                Rails.root.join(
+                  'spec', 'fixtures',
+                  'ial2_test_credential_no_liveness.yml'
+                ),
+              )
+              attach_selfie(
+                Rails.root.join(
+                  'spec', 'fixtures',
+                  'ial2_test_credential_no_liveness.yml'
+                ),
+              )
+
+              submit_images
+
+              h1_error_message = strip_tags(
+                t('errors.doc_auth.selfie_not_live_or_poor_quality_heading'),
+              )
+              expect(page).to have_content(h1_error_message)
+
+              body_error_message = strip_tags(t('doc_auth.errors.alerts.selfie_not_live'))
+              expect(page).to have_content(body_error_message)
+
+              click_try_again
+              expect(page).to have_current_path(idv_document_capture_path)
```

Comment:
I think we need to check the text on the upload page?

Reply:
That's a good point. I think this case is similar to the above and I should have added a comment here like I did above of `# inline error to be fixed in lg-12999`.

I will wait to see your thoughts of the above case and adjust this to be the same (whether I just add an expectation of what we see now that we know is wrong, or add in the above comment as a marker to come back in lg-12999).

I'm sorry I didn't do a good job of communicating this. I tried to mention in our meeting about the two tickets and how they interacted it but I'm afraid I didn't talk about it in a clear way.